qiling/os/posix/syscall/stat.py

```diff
@@ -163,102 +163,109 @@ def create_stat64_struct(ql, info):
-        fstat64_buf += ql.pack32(0)
-        fstat64_buf += ql.pack64(info.st_blocks)
+          stat64_buf = ql.pack32(info.st_dev)
+        stat64_buf += b'\x00' * 12 
+        stat64_buf += ql.pack64(info.st_ino)
+        stat64_buf += ql.pack32(info.st_mode)
+        stat64_buf += ql.pack32(info.st_nlink)
+        stat64_buf += ql.pack32(ql.os.uid)
+        stat64_buf += ql.pack32(ql.os.gid)
+        stat64_buf += ql.pack32(info.st_rdev)
+        stat64_buf += b'\x00' * 12
+        stat64_buf += ql.pack64(info.st_size)
+        stat64_buf += ql.pack32(int(info.st_atime))
+        stat64_buf += ql.pack32(0)
+        stat64_buf += ql.pack32(int(info.st_mtime))
+        stat64_buf += ql.pack32(0)
+        stat64_buf += ql.pack32(int(info.st_ctime))
+        stat64_buf += ql.pack32(0)
+        stat64_buf += ql.pack32(info.st_blksize)
+        stat64_buf += ql.pack32(0)
+        stat64_buf += ql.pack64(info.st_blocks)
     elif ql.archtype == QL_ARCH.ARM:
-        # pack fstatinfo
+        # pack statinfo
         if ql.platform == QL_OS.MACOS:
-            fstat64_buf = ql.pack64s(info.st_dev)
+            stat64_buf = ql.pack64s(info.st_dev)
         else:
-            fstat64_buf = ql.pack64(info.st_dev)
-        fstat64_buf += ql.pack32(0)
-        fstat64_buf += ql.pack32(info.st_ino)
-        fstat64_buf += ql.pack32(info.st_mode)
-        fstat64_buf += ql.pack32(info.st_nlink)
-        fstat64_buf += ql.pack32(info.st_uid)
-        fstat64_buf += ql.pack32(info.st_gid)
-        fstat64_buf += ql.pack64(info.st_rdev)  # ?? fstat_info.st_rdev
-        fstat64_buf += ql.pack64(0)
-        fstat64_buf += ql.pack64(info.st_size)
-        fstat64_buf += ql.pack64(info.st_blksize)  # ?? fstat_info.st_blksize
-        fstat64_buf += ql.pack64(info.st_blocks)  # ?? fstat_info.st_blocks
-        fstat64_buf += ql.pack64(int(info.st_atime))
-        fstat64_buf += ql.pack64(int(info.st_mtime))
-        fstat64_buf += ql.pack64(int(info.st_ctime))
-        fstat64_buf += ql.pack64(info.st_ino)
+            stat64_buf = ql.pack64(info.st_dev)
+        stat64_buf += ql.pack32(0)
+        stat64_buf += ql.pack32(info.st_ino)
+        stat64_buf += ql.pack32(info.st_mode)
+        stat64_buf += ql.pack32(info.st_nlink)
+        stat64_buf += ql.pack32(info.st_uid)
+        stat64_buf += ql.pack32(info.st_gid)
+        stat64_buf += ql.pack64(info.st_rdev)  # ?? stat_info.st_rdev
+        stat64_buf += ql.pack64(0)
+        stat64_buf += ql.pack64(info.st_size)
+        stat64_buf += ql.pack64(info.st_blksize)  # ?? stat_info.st_blksize
+        stat64_buf += ql.pack64(info.st_blocks)  # ?? stat_info.st_blocks
+        stat64_buf += ql.pack64(int(info.st_atime))
+        stat64_buf += ql.pack64(int(info.st_mtime))
+        stat64_buf += ql.pack64(int(info.st_ctime))
+        stat64_buf += ql.pack64(info.st_ino)
     elif ql.ostype == QL_OS.MACOS:
-        fstat64_buf = ql.pack32(0x0)              # st_dev            32byte
```

Comment:
macos 10.15, we need some hack to make this st_dev to 0x0

can you please add it back

Reply:
Done